pkg/spec/spec.go

```diff
@@ -11,9 +11,19 @@ type PersistentVolume struct {
 	Size                             string `json:"size"`
 }
 
-type Service struct {
-	Name                    string `json:"name,omitempty"`
-	api_v1.ServiceSpec      `json:",inline"`
+type ServicePortMod struct {
+	api_v1.ServicePort `json:",inline"`
+	Endpoint           string `json:"endpoint"`
```

Comment:
it would be nice to have comments in new fields that we are adding describing what they are for